src/services/inlayHints.ts

```diff
@@ -137,10 +137,18 @@ namespace ts.InlayHints {
 
             const typeDisplayString = printTypeInSingleLine(declarationType);
             if (typeDisplayString) {
+                const isVariableNameMatchesType = !preferences.includeInlayVariableTypeHintsWhenTypeMatchesName && variableDeclarationNameMatchesTypeString(decl.name.getText(), typeDisplayString);
+                if (isVariableNameMatchesType) {
+                    return;
+                }
                 addTypeHints(typeDisplayString, decl.name.end);
             }
         }
 
+        function variableDeclarationNameMatchesTypeString(declarationName: string, typeDisplayString: string) {

```

Comment:
I don’t like functions with highly specific names but very general functionality. You can delete this and use `equateStringsCaseInsensitive`.